packages/workbox-cacheable-response/CacheableResponsePlugin.mjs

```diff
@@ -0,0 +1,44 @@
+/*
+ Copyright 2016 Google Inc. All Rights Reserved.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+     http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+import './_version.mjs';
+
+/**
+ * A class implementing the `cacheWillUpdate` lifecycle callback. This makes it
+ * easier to add in cacheability checks to requests made via Workbox's built-in
+ * strategies.
+ *
+ * @memberof workbox.cacheableResponse
+ */
+class CacheableResponsePlugin {
+  /**
+   * @param {CacheableResponse} cacheableResponse A configured
```

Comment:
Why is this being passed in? This should just either extend or construct the value.

Will be a horrible API otherwise:

```javascript
workbox.cacheable.CacheableResponsePlugin(
  new workbox.cacheable.CacheableResponse({...})
);
```

Reply:
Right, that's why I was asking about v3 best practices for this in the Slack channel.

I'm happy to change it to take the same parameters as the `CacheableResponse` class and construct an internal instance under the hood. That seems cleaner to me, too.

I based this code on what's being done in https://github.com/GoogleChrome/workbox/blob/v3/packages/workbox-background-sync/QueuePlugin.mjs#L28, and maybe the same change should be considered there? CC: @philipwalton

Reply:
I'm fine with changing the API to have the plugins class take the same params as the underlying class and just expose the plugin methods. The main downside of that is there's code and documentation duplication.

My primary concern with the old model (using class inheritance) was it made it easy for accidental naming collisions, and I don't think we should go back to that.